substrate/frame/support/src/traits/tokens/fungible/regular.rs

```diff
@@ -408,17 +410,21 @@ pub trait Balanced<AccountId>: Inspect<AccountId> + Unbalanced<AccountId> {
 	/// This is just the same as burning and issuing the same amount and has no effect on the
 	/// total issuance.
 	fn pair(amount: Self::Balance) -> (Debt<AccountId, Self>, Credit<AccountId, Self>) {
-		(Self::rescind(amount), Self::issue(amount))
+		// Create credit and debt in-line to avoid overflow or underflow which could occur when
+		// calling `issue` and `rescind` directly.
+		let credit = Imbalance::<Self::Balance, Self::OnDropCredit, Self::OnDropDebt>::new(amount);
+		let debt = Imbalance::<Self::Balance, Self::OnDropDebt, Self::OnDropCredit>::new(amount);
```

Comment:
with this approach you still have a problem with amounts which will cause the overflow.
you might want to find a delta here, and use min from delta and amount, and pass that min to rescind and issue, plus updated docs

Reply:
Could you clarify how can this implementation can still cause overflow?

Reply:
check impl of rescind for example, it does account the total issuance. so may be better solution would be 
```
let amount = amount.min(Self::total_issuance()));
(Self::rescind(amount), Self::issue(amount))
```

Reply:
My gut feeling is to just make the function do what it says it does and create the pair with the amounts passed to it, and assume the caller knows what they are doing. 

Otherwise the max amount able to be created with `pair` is limited by two bounds, the delta from current issuance to the total issuance AND the delta from the current issuance to zero. This seems confusing and could lead to weird behavior. This adds a lot of complexity to the method which is kinda difficult to reason about, and maybe not what the caller wants.

Reply:
sorry, my example is wrong.
I think it should be
```
let total_issuance = Self::total_issuance(); // this max value we can account if only credit is dropped
let total_issuance_left = total_issuance.max(Balance::MAX) - total_issuance.min(Balance::MAX) // this max value we can account if only debt is dropped
let amount = total_issuance.min(total_issuance_left);
(Self::rescind(amount), Self::issue(amount))
``` 
if you check `rescind` and `issue` they behave in a similar way, they rescind or issue the `amount` at best.

Reply:
I couldn't find any value akin to `Balance::MAX` available for use.

This is `Balance`:
```rust
/// Simple amalgamation trait to collect together properties for a Balance under one roof.
pub trait Balance:
	AtLeast32BitUnsigned
	+ FullCodec
	+ Copy
	+ Default
	+ Debug
	+ scale_info::TypeInfo
	+ MaxEncodedLen
	+ Send
	+ Sync
	+ MaybeSerializeDeserialize
	+ 'static
```

Any ideas? Maybe we need a new method on `Inspect`?

Reply:
you still can do it with saturation_sub/add and calculating the deltas for rescind and issue, and using the min from both